### PR TITLE
feat: change modal styles prop interface

### PR DESCRIPTION
**Pull request by samobasquiat**
# Summary

The modal style prop interface is changed to make the modal design more accessible.
Also, Divider got a props called color.

# Checklist:

- [x] I have performed a self-review of my code

### CODE REVIEW

**Review by RanGojo — APPROVED** [review]

File: widget/ui/src/icons/ExitToApp.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import type { SvgIconPropsWithChildren } from '../components/SvgIcon';
+
+import React, { createElement } from 'react';
+
+import { SvgIcon } from '../components/SvgIcon';
+
+function SvgExitToApp(props: SvgIconPropsWithChildren) {
```
Comment (RanGojo) [inline widget/ui/src/icons/ExitToApp.tsx:7]: I think `SvgExit` is better

File: widget/ui/src/components/Toast/Toast.tsx [hunk 2]
```diff
@@ -14,23 +14,26 @@ import {
   ToastContentContainer,
 } from './Toast.styles';
 
-const DEFAULT_HIDE_DURATION = 3_000;
-
 export const Toast = (props: ToastProps) => {
-  const { id, horizontal, autoHideDuration = DEFAULT_HIDE_DURATION } = props;
+  const { id, autoHideDuration, onClose } = props;
```
Comment (RanGojo) [inline widget/ui/src/components/Toast/Toast.tsx:18]: I didn't understand why we have removed default value here.

File: widget/ui/src/components/Toast/Toast.Provider.tsx [hunk 3]
```diff
@@ -62,14 +53,19 @@ export const ToastProvider = (props: PropsWithChildren<ProviderPropTypes>) => {
...
+              ({ containerStyle, horizontal, vertical, ...toast }) => (
+                <ToastContainer
+                  key={toast.id}
+                  style={containerStyle}
+                  horizontal={horizontal || 'center'}
+                  vertical={vertical || 'top'}>
+                  <Toast {...toast} key={toast.id} />
+                </ToastContainer>
+              )
+            )}
```
Comment (RanGojo) [inline widget/ui/src/components/Toast/Toast.Provider.tsx:67]: If we have two parallel toast with different vertical props, does it work well?

File: widget/ui/src/components/Divider/Divider.tsx [hunk 4]
```diff
@@ -9,6 +9,14 @@ const DEFAULT_SIZE = 12;
...
 }: PropTypes) {
-  return <DividerContainer size={size} direction={direction} />;
+  const customCss = color
+    ? {
+        backgroundColor: color.startsWith('$') ? color : `$${color}`,
+      }
+    : {
+        backgroundColor: 'transparent',
+      };
+  return <DividerContainer size={size} direction={direction} css={customCss} />;
```
Comment (RanGojo) [inline widget/ui/src/components/Divider/Divider.tsx:21]: isn't possible to change it to a prop for css or style instead? current code is not very clean.

File: widget/ui/src/components/Toast/Toast.tsx [hunk 5]
```diff
@@ -14,23 +14,26 @@ import {
...
   const { removeToast } = useToast();
   useEffect(() => {
-    const timer = setTimeout(() => {
-      removeToast(id);
-    }, autoHideDuration);
+    let cleanup;
+
+    if (autoHideDuration) {
```
Comment (RanGojo) [inline widget/ui/src/components/Toast/Toast.tsx:23]: Please check animation / close problem in dapp.

**Review by RanGojo — CHANGES_REQUESTED** [review]

File: widget/ui/src/components/Toast/Toast.tsx (hunk 2 above)
Comment (samobasquiat) [reply widget/ui/src/components/Toast/Toast.tsx:18]: If it has autoHideDuration it is removed if not the user should handle it by `removeToast` manually. in dapp some toasts have` autoHide={false}`

we can also consider having 2 props: `autoHide` & `autoHideDuration`. but currently, user can set it by `autoHideDuration`

File: widget/ui/src/components/Toast/Toast.Provider.tsx (hunk 3 above)
Comment (samobasquiat) [reply widget/ui/src/components/Toast/Toast.Provider.tsx:67]: I changed the structure to handle these situations.

File: widget/ui/src/components/Divider/Divider.tsx (hunk 4 above)
Comment (samobasquiat) [reply widget/ui/src/components/Divider/Divider.tsx:21]: I wanted to have a similar style to the Typography component. however, changed.

File: widget/ui/src/components/Toast/Toast.tsx (hunk 5 above)
Comment (samobasquiat) [reply widget/ui/src/components/Toast/Toast.tsx:23]: I think the animation should be handled in another task. But I fixed the overlap issue.

File: widget/ui/src/components/Toast/Toast.Provider.tsx [hunk 6]
```diff
@@ -18,58 +19,152 @@ const ToastContext = createContext<ProviderContext | undefined>(undefined);
...
+            {centerBottomToasts.length > 0 && (
+              <ToastContainer
+                style={centerBottomToasts[0].containerStyle}
+                position="center-bottom">
+                {centerBottomToasts.map((toast) => (
+                  <Toast {...toast} key={toast.id} />
+                ))}
+              </ToastContainer>
+            )}
+          </>,
```
Comment (RanGojo) [inline widget/ui/src/components/Toast/Toast.Provider.tsx:167]: You could refactor  it like this:

```ts
const renderToastContainer = (toasts: ToastProps[], position: ToastPosition) => (
  <>
    {toasts.length > 0 && (
      <ToastContainer
        style={toasts[0].containerStyle}
        position={position}
      >
        {toasts.map((toast) => (
          <Toast {...toast} key={toast.id} />
        ))}
      </ToastContainer>
    )}
  </>
);

// Usage
<>
  {renderToastContainer(rightTopToasts, 'right-top')}
  {renderToastContainer(leftTopToasts, 'left-top')}
  {renderToastContainer(centerTopToasts, 'center-top')}
  {renderToastContainer(rightBottomToasts, 'right-bottom')}
  {renderToastContainer(leftBottomToasts, 'left-bottom')}
  {renderToastContainer(centerBottomToasts, 'center-bottom')}
</>```

**Review by RanGojo — APPROVED** [review]

**Pull request merged**